kernel/work.c

```diff
@@ -761,6 +761,10 @@ void k_work_queue_start(struct k_work_q *queue,
 		k_thread_name_set(&queue->thread, cfg->name);
 	}
 
+	if (cfg->essential) {
+		queue->thread.base.user_options |= K_ESSENTIAL;
+	}
```

Comment:
Perhaps there is a reason that I am missing, but why not conditionally pass K_ESSENTIAL as the _options_ parameter in the call to _k_thread_create()_ above?  That is something like, ...

`    k_thread_create(&queue->thread, ...., prio, cfg->essential ? K_ESSENTIAL : 0, K_FOREVER);`

That would eliminate some messing around with k_thread internals.

Reply:
since it is acceptable to pass `cfg=NULL` this can cause a crash, i modified the condition to also check if `cfg != NULL`